src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
Could this safely be changed to VersionBitsState(...) == THRESHOLD_LOCKED_IN || VersionBitsState(...) == THRESHOLD_ACTIVE rather than just VersionBitsState(...) == THRESHOLD_ACTIVE to speed things up a bit? As far as I can tell, keeping this as is would delay BIP141 enforcement by an additional full difficulty adjustment period, and I'm not sure what the benefit would be.